--- e2fsprogs/misc/create_inode_libarchive.c
+++ e2fsprogs/misc/create_inode_libarchive.c
@@ -18,14 +18,14 @@
 #include "support/nls-enable.h"
 
 #if (!(defined(CONFIG_DLOPEN_LIBARCHIVE) || defined(HAVE_ARCHIVE_H)) || \
-     defined(CONFIG_DIABLE_LIBARCHIVE))
+     defined(CONFIG_DISABLE_LIBARCHIVE))
 
 /* If ./configure was run with --without-libarchive, then only
  * __populate_fs_from_tar() remains in this file and will return an error. */
-errcode_t __populate_fs_from_tar(ext2_filsys, ext2_ino_t, const char *,
-                                 ext2_ino_t, struct hdlinks_s *,
-                                 struct file_info *,
-                                 struct fs_ops_callbacks *) {
+errcode_t __populate_fs_from_tar(ext2_filsys fs, ext2_ino_t root_ino, const char *source_tar,
+                                 ext2_ino_t root, struct hdlinks_s *hdlinks,
+                                 struct file_info *target,
+                                 struct fs_ops_callbacks *fs_callbacks) {
   com_err(__func__, 0,
           _("you need to compile e2fsprogs without --without-libarchive"
             "be able to process tarballs"));
